### ISSUE TITLE
Enable / disable auto-save

### ISSUE THREAD

**Issue by Anguel** [simsalabim/sisyphus#117]
Hi,

I need to allow the user to enable / disable auto-saving of the form through a button on the page. Can this be achieved through onSave() or is there any way to dynamically register / unregister a form with Sisyphus? I tried calling manuallyReleaseData() in onSave() but this does not seem to ba a good idea and slows down the script.
Thanks.

Anguel

**Comment by gabrieljenik** [comment]
I have already some code for that. Will send it to you during the day.

**Comment by Anguel** [comment]
Cool, thanks. I was thinking about setting some variable in local storage and then reloading the whole page but this is probably too much trouble as my form is on tabs :(

**Comment by gabrieljenik** [comment]
Check this PR:
https://github.com/simsalabim/sisyphus/pull/114/commits

**Comment by gabrieljenik** [comment]
Also check this code:
https://github.com/gabrieljenik/sisyphus/blob/master/ondemmand_sisyphus.js

**Comment by Anguel** [comment]
Very nice! I hope this will be merged into Sisyphus soon. Thank you once again.

**Comment by Anguel** [comment]
@gabrieljenik : Once again, thank you very much. Now I have a autosave checkbox which sets a variable that is read in your extremely useful onBeforeSave() function and actually defines if autosave will take place or not. Unchecking the autosave checkbox will also call $('#myform').sisyphus().manuallyReleaseData(). Additionally, if the user has disabled the checkbox but decides to reenable autosaving at some point and has entered some data in between I call $('#myform').sisyphus().saveAllData() when he checks the checkbox. Do you think this is safe to do?

**Comment by gabrieljenik** [comment]
Why do you need to release the data? IN regard to "safety" I don't see any thread... do you see something to fear about?

**Comment by Anguel** [comment]
Actually, I am adding the checkbox for users who want more "privacy", maybe they are on some public PC and do not want all their form data to stay on that PC, therefore I release the data when they uncheck the box. I asked if it is ok to call saveAllData() directly because I don't have much experience with JavaScript and I don't want to break anything :)
BTW, another problem is when the browser is in "safe browsing mode" - browser says it supports local storage and data is saved but will get lost when the private tab is closed :( There seems to be no way to actually tell if browser is in private mode AFAIK and warn the user...

**Comment by gabrieljenik** [comment]
Calling saveAllData() seems good idea.
About private mode, wouldn't know about that. May be you can put some kind of warning message or bubble message near the checkbox.

**Comment by Anguel** [comment]
Yes, I will put a warning. Not the best solution, but at least the user cannot sue me for losing his data ;)))
